src/App.tsx

```diff
@@ -14,11 +15,112 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[],
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: [],
+  };
+
+  clearActivator = false;
+
+  addGood = (good: string) => {
+    this.setState((currentState) => ({
+      selectedGoods: [...currentState.selectedGoods, good],
+    }));
+  };
+
+  removeGood = (good: string) => {
+    this.setState((currentState) => ({
+      selectedGoods: currentState.selectedGoods.filter(item => item !== good),
+    }));
+  };
+
+  getGoods = () => {
+    const result = this.stringCreator();
+
+    return result;
+  };
+
+  stringCreator = () => {
+    const { selectedGoods } = this.state;
+    let result;
+
+    if (selectedGoods.length === 1) {
+      result = `${selectedGoods[0]} is selected`;
+    } else {
+      const slicedArray = selectedGoods.slice(0, selectedGoods.length - 1);
+
+      result = `${slicedArray.join(', ')} and ${selectedGoods[selectedGoods.length - 1]} are selected`;
+    }
+
+    return result;
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <h1
+          className="title"
+        >
+          Selected goods:
+        </h1>
+        <div className="container">
+          {selectedGoods.length === 0 ? <div>No goods selected</div> : <div className="selected-good">{this.getGoods()}</div>}
+          <button
+            type="button"
+            className={classNames('clear-button', { 'clear-button--active': selectedGoods.length > 0 })}
+            onClick={() => {
+              this.setState({ selectedGoods: [] });
+            }}
+          >
+            X
+          </button>
+        </div>
+
+        <ul className="list">
+          <div className="header-container">
+            <div className="header-goods">
+              Goods
+            </div>
+
+            <div className="header-action">
+              Action
+            </div>
+          </div>
+          {goodsFromServer.map(good => (
```

Comment:
```suggestion
        <div className="header-container">
          <div className="header-goods">
            Goods
          </div>
          <div className="header-action">
            Action
          </div>
        </div>
        <ul className="list">
          {goodsFromServer.map(good => (
```
всередині `<ul>` може бути лише `<li>`. `div` всередині `ul` - не валідний код